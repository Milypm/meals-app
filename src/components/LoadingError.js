import FadeLoader from 'react-spinners/FadeLoader';
import { css } from '@emotion/react';
import '../styles/home.css';

const override = css`
  display: block;
  margin: 0 auto;
  border-color: red;
`;

export const Loading = () => {
  const loading = true;
  return (
    <div className="loading-div">
      <h3>Loading</h3>
      <FadeLoader color="rgb(219, 135, 96)" loading={loading} css={override} size={850} />
    </div>
  );
};
export const Error = () => (
  <div className="error-div">
    <h3>Sorry, we couldn&apos;t display the information :( Try again please</h3>
  </div>
);
