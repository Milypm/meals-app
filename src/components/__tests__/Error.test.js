import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Error } from '../LoadingError';

test('Error component renders correctly', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Error />
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
