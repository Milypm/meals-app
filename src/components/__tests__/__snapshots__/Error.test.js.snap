// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component renders correctly 1`] = `
<div
  className="error-div"
>
  <h3>
    Sorry, we couldn't display the information :( Try again please
  </h3>
</div>
`;
